api/v1alpha1/httproute_types.go

```diff
@@ -115,15 +116,21 @@ type HTTPRouteMatch struct {
 	// Default: "/"
 	Path *string `json:"path" protobuf:"bytes,2,opt,name=path"`
 
-	// HeaderType defines the semantics of the `Header` matcher.
+	// HeaderMatchType defines the semantics of the `Header` matcher.
+	//
+	// Support: core (Exact)
+	// Support: custom (ImplementationSpecific)
```

Comment:
I'm not quite sure what "custom" means here, but this does match what we already have for path type matching. I've created an issue to track documenting these different levels: https://github.com/kubernetes-sigs/service-apis/issues/195